LFS-709: Subject selector improvements; LFS-721:Unable to create Tumor form in a specific situation; LFS-688: UI anomaly when having multiple tumors with the same name

veronikaslc opened the pull request:
[LFS-709](https://phenotips.atlassian.net/browse/LFS-709): Subject selector improvements - 3/4 subtasks:
* [LFS-711](https://phenotips.atlassian.net/browse/LFS-711): Remove Select a <SubjectType> from the dialog body
* [LFS-708](https://phenotips.atlassian.net/browse/LFS-708): Display the full id for available subjects
* [LFS-720](https://phenotips.atlassian.net/browse/LFS-720): Improve error message

[LFS-721](https://phenotips.atlassian.net/browse/LFS-721):Unable to create Tumor form in a specific situation

[LFS-688](https://phenotips.atlassian.net/browse/LFS-688): UI anomaly when having multiple tumors with the same name

Commit:
LFS-709: Subject selector improvements
LFS-711: Remove Select a <SubjectType> from the dialog body

Commit:
LFS-709: Subject selector improvements
LFS-708: Display the full id for available subjects

Commit:
LFS-709: Subject selector improvements
Fixed artefact typo

marta- reviewed (approved):
Looks good. ~Also fixes [LFS-688](https://phenotips.atlassian.net/browse/LFS-688) _UI anomaly when having multiple tumors with the same name_ .~

sashaandjic commented:
- 711 - tested; all good 
- 709 - the full id is displayed, however the new problem is introduced. Steps to reproduce:
- create 3 patients ( P1, P2, P3)
- create 3 same tumors and assign them to each patient ( P1/liver, P2/liver. P3/liver)
- create tumor form for P1/liver
- create tumor form for P2/liver - the error will be shown **"Tumor liver already has 1 Tumors Data form(s) filled out."**

sashaandjic commented:
Another problem:
- create a patient (P1) 
- create tumors and assign the to patient (P1/liver)
- create tumor region 1 and assign it to P1/liver
- try to create another tumor region 1 for P1/liver - it will fail (**correct**), however the error message is incomplete "Subject 1 already exist for parent liver". It should state "Tumor region 1 already exist for tumor P1/liver."

marta- committed:
LFS-721: Unable to create Tumor form in a specific situation

marta- committed:
LFS-709: Subject selector improvements
[Addressing comments] Clearer error message when a child subject id already exists for the selected parent when creating a new child

marta- committed:
LFS-711: Remove Select a from the dialog body
Prevent the default material table title from being displayed

marta- committed:
LFS-688: UI anomaly when having multiple tumors with the same name

sashaandjic reviewed (approved):
tested
